src/rpc_methods.rs

```diff
@@ -186,30 +185,31 @@ pub struct PeerInfo {
     /// The total bytes received aggregated by message type
     pub bytesrecv_per_msg: LinearMap<String, u64>,
 }
+
 impl PeerInfo {
-    pub fn into_address(self) -> Result<Address, Error> {
-        let services = ServiceFlags::consensus_decode(&mut std::io::Cursor::new(hex::decode(
-            &self.services,
-        )?))?;
+    pub fn into_address(self) -> Result<Address, PeerAddressError> {
+        let decoded_services = hex::decode(&self.services)
+            .map_err(|error| PeerAddressError::InvalidHex { string: self.services.clone(), error, })?;
+        let services = ServiceFlags::consensus_decode(&mut std::io::Cursor::new(decoded_services))
+            .map_err(|error| PeerAddressError::ConsensusDecode { string: self.services.clone(), error, })?;
         if let Ok(sock_addr) = self.addr.parse() {
             Ok(Address::new(&sock_addr, services))
         } else {
             let mut addr_split = self.addr.split(":");
             let host = addr_split
                 .next()
-                .ok_or_else(|| anyhow!("Invalid Peer Address: {}", self.addr))?;
+                .expect("std::str::split() is an empty iterator which should never happen");
```

Comment:
Normally I would prefix these kinds of error messages with `UNREACHABLE!: `

Reply:
Is this some more widely-used pattern or your own preference?

Reply:
This is a personal preference, I'm not aware of a specific widely used convention for this, however it is generally good practice to have a consistent convention around code that is supposed to be unreachable.
If you want the error message to be consistent with the rust `unreachable!` macro, you can also prefix it with `internal error: entered unreachable code: `.

Reply:
Yeah, I like the idea of making it easy to identify. `internal error: entered unreachable code: ` is pretty long, I think `error: entered unreachable code: ` would be sufficient, `UNREACHABLE!` is not as nice for the users if for some insane reason it stays in the code.

Actually, thinking about it, maybe the compiler can prove that this code is unreachable, so we can use [`dont_panic`](https://crates.io/crates/dont_panic), I'm going to try it.

Reply:
Sadly, the code seems to be too complicated for rustc to prove it's unreachable.

Reply:
if you are confident its unreachable you can always use:
```rust
unsafe { std::hint::unreachable_unchecked() }
```
however it will result in undefined behavior if it turns out to be wrong 🙃

Reply:
That's something I definitely wouldn't do in *this* case. The whole reason rustc can't prove it is it internally uses pattern API which makes it quite complicated. I guess one day in the future we can rewrite it to use `split_once` but that will take a lot of waiting. (Rust stabilizing *and* Debian getting that version into stable.)